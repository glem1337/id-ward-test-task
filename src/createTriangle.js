import createModal from './createModal';

/**
 * Create triangle
 */
const createTriangle = () => {
  const triangle = document.createElement('div');

  triangle.classList.add('plugin-triangle');

  triangle.onclick = createModal;

  document.body.appendChild(triangle);

  return triangle;
};

export default createTriangle;
