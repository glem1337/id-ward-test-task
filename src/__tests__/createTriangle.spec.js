import createTriangle from '../createTriangle';

describe('createTriangle()', () => {
  const mockedTriangle = {
    classList: {
      add: jest.fn(),
    },
  };
  const mockedBody = {
    appendChild: jest.fn(),
  };

  Object.defineProperty(document, 'createElement', { value: jest.fn(() => mockedTriangle) });
  Object.defineProperty(document, 'body', {
    value: mockedBody,
  });

  it('matches snapshot', () => {
    expect(createTriangle()).toMatchSnapshot();
  });

  it('should creates triangle', () => {
    createTriangle();

    expect(document.createElement).toHaveBeenCalledWith('div');
    expect(mockedTriangle.classList.add).toHaveBeenCalledWith('plugin-triangle');
    expect(mockedBody.appendChild).toHaveBeenCalledWith(mockedTriangle);
  });
});
