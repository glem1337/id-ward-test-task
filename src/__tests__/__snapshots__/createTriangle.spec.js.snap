// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createTriangle() matches snapshot 1`] = `
Object {
  "classList": Object {
    "add": [MockFunction] {
      "calls": Array [
        Array [
          "plugin-triangle",
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    },
  },
  "onclick": [Function],
}
`;
